src/index.ts

```diff
@@ -95,38 +96,22 @@ const calculateDepreciation = ({
     }]
   }
 
-  for (let index = 0; index < totalDepreciationYears; index++) {
-    // Current year
-    if (index === 0) {
-      const monthsLeftInFirstYear = MONTHS_IN_YEAR - purchaseMonth + 1;
-      const result = calculate(purchaseAmount, totalDepreciationYears, endAmount, monthsLeftInFirstYear);
-      results.push({
-        year: purchaseYear,
-        depreciationMonths: monthsLeftInFirstYear,
-        ...result,
-      });
-      endAmount = result.endAmount;
-
-      // If item is purchased in Jan, monthsLeftInLastYear is 0
-      // If item is purchased in Dec, monthsLeftInLastYear is 11
-      monthsLeftInLastYear = MONTHS_IN_YEAR - monthsLeftInFirstYear;
-    } else { // Future years
-      const result = calculate(purchaseAmount, totalDepreciationYears, endAmount);
-      results.push({
-        year: purchaseYear + index,
-        depreciationMonths: MONTHS_IN_YEAR,
-        ...result,
-      });
-      endAmount = result.endAmount;
-    }
-  }
+  const parts: number = purchaseMonth > 1 ? totalDepreciationYears + 1 : totalDepreciationYears;
+
+  let monthsInEachYear: number[] = Array(parts).fill(MONTHS_IN_YEAR);
+  //months in fisrt year
+  monthsInEachYear[0] = MONTHS_IN_YEAR - purchaseMonth + 1;
+  //months in last year
+  monthsInEachYear[monthsInEachYear.length - 1] = purchaseMonth - 1 > 0 ? purchaseMonth - 1 : MONTHS_IN_YEAR;
```

Comment:
```suggestion
  monthsInEachYear[monthsInEachYear.length - 1] = purchaseMonth > 1 ? purchaseMonth - 1 : MONTHS_IN_YEAR;
```

In [line 99](https://github.com/kontist/asset-depreciation-calculator/pull/4/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R99), we `purchaseMonth > 1` instead of `purchaseMonth - 1 > 0`. I think it’s easier to understand if we use the same comparison here.

Reply:
Ok, I'm agree. :-)